cpp/log/cert_verifier.h

```diff
@@ -0,0 +1,58 @@
+/* -*- mode: c++; indent-tabs-mode: nil -*- */
+#ifndef CERT_VERIFIER_H
+#define CERT_VERIFIER_H
+
+#include <memory>
+#include <openssl/asn1.h>
+#include <openssl/bio.h>
+#include <openssl/cms.h>
+
+#include "base/macros.h"
+#include "log/cert.h"
+#include "util/openssl_util.h"  // for LOG_OPENSSL_ERRORS
+#include "util/status.h"
+
+namespace cert_trans {
+
+class CertVerifier {
```

Comment:
Having `CertChecker` and `CertVerifier` could get a bit confusing, how about `CMSVerifier`, (although maybe that's not so good as this doesn't [currently] verify the signature)?